### ISSUE TITLE
Crash on update favicons after deleting windows when windows are deleted

### ISSUE THREAD

**Issue by jgkamat** [qutebrowser/qutebrowser#3781]
On the latest master, there is a crash when updating the favicon state if tabs_are_windows is set after deleting a window. Originally reported here: https://github.com/qutebrowser/qutebrowser/pull/3680#issuecomment-376637166

I did a bisect, and I think e169e2165da7892c0b9731a94481aebbc867df26 is the culprit.

To reproduce, open a window `:open -w`, close the window, then turn off favicons.

It seems like we are not preparing properly for `self.widget` to be deleted (we don't hook window close or anything like that), and that's why this is happening.

```
00:20:35 ERROR: Uncaught exception
Traceback (most recent call last):
  File "/home/jay/Code/qutebrowser/qutebrowser/mainwindow/tabbedbrowser.py", line 143, in _on_config_changed
    self._update_favicons()
  File "/home/jay/Code/qutebrowser/qutebrowser/mainwindow/tabbedbrowser.py", line 534, in _update_favicons
    for i, tab in enumerate(self.widgets()):
  File "/home/jay/Code/qutebrowser/qutebrowser/mainwindow/tabbedbrowser.py", line 171, in widgets
    for i in range(self.widget.count()):
RuntimeError: wrapped C/C++ object of type TabWidget has been deleted
```

**Comment by jgkamat** [comment]
This is naive, but it does fix the issue. I'm not sure how TabbedBrowser actually handles window deletion though, so I'm not sure if it will cause more problems :P 

```
diff --git a/qutebrowser/mainwindow/tabbedbrowser.py b/qutebrowser/mainwindow/tabbedbrowser.py
index cedb0cf33..414e96b21 100644
--- a/qutebrowser/mainwindow/tabbedbrowser.py
+++ b/qutebrowser/mainwindow/tabbedbrowser.py
@@ -133,6 +133,7 @@ class TabbedBrowser(QWidget):
         self.default_window_icon = self.widget.window().windowIcon()
         self.private = private
         config.instance.changed.connect(self._on_config_changed)
+        self.widget.destroyed.connect(self.deleteLater)
 
     def __repr__(self):
         return utils.get_repr(self, count=self.widget.count())
```

**Comment by The-Compiler** [comment]
I think I found a better fix in c7e5033eaa3580d5044a3393dad39e3fb53dc418 (nevermind the accidental `misc.feature` change :blush:).

I've seen some similar crash reports recently (haven't looked at them or opened issues yet), makes me wonder whether the same issue was to blame!
